src/slam_karto.cpp

```diff
@@ -280,8 +293,21 @@ SlamKarto::SlamKarto() :
     mapper_->setParamUseResponseExpansion(use_response_expansion);
 
   // Set solver to be used in loop closure
-  solver_ = new SpaSolver();
-  mapper_->SetScanSolver(solver_);
+  try
+  {
+    std::stringstream ss;
+    ss<<"karto_plugins::"<<solver_type_;
+    solver_ = solver_loader_.createInstance(ss.str());
+    ROS_INFO_STREAM("Loaded "<<ss.str()<<" solver plugin");
+  }
+  catch(pluginlib::PluginlibException& ex)
+  {
+    ROS_ERROR("The plugin failed to load for some reason. Error: %s", ex.what());
```

Comment:
Add name of the plugin that failed to load here:
The plugin %s failed to load...
Add fall-back behavior here?